features/other_a/pseuds.feature

```diff
@@ -117,3 +117,20 @@ Scenario: Manage pseuds - add, edit
     And I should see "My new fancy name (editpseuds)"
     And I should see "I wanted to add another fancy name"
     And I should not see "My new name (editpseuds)"
+
+Scenario: Comments reflect pseud changes immediately
+
+  Given the work "Interesting"
+    And I am logged in as "myself"
+    And I add the pseud "before"
+  When I set up the comment "Wow!" on the work "Interesting"
+    And I select "before" from "comment[pseud_id]"
+    And I press "Comment"
+    And I view the work "Interesting" with comments
+  Then I should see "before (myself)" within "h4.heading.byline"
+
+  When I change the pseud "before" to "after"
+    And it is currently 10 seconds from now
```

Comment:
I added this because running this and the tests in `user_rename.feature` made this fail. I suspect something weird is happening in the cache (running `flush_all` from telnet made things happier), but if I add breakpoints and poke around, whatever timing issue it is doesn't occur. Seems a bit hacky to me, but also caching is hard :shrug:

Reply:
Yeah, it's because the cache is expired when `updated_at` changes, but `updated_at` is only accurate down to the second. So if you create a pseud and then change the pseud name within the same second, the code to expire the comment cache won't work without some kind of artificial delay.


Eventually I'd like to increase the precision of all timestamps, so that we don't have to keep adding delays to make the tests pass more consistently. But practically speaking, most people aren't going to experience that in production, so changing the precision on all those timestamps

Reply:
Thanks as always for the thorough explanation ticking! I've made your suggested change, and the tests are much happier now :tada: